micro-rdk/src/esp32/entry.rs

```diff
@@ -193,46 +146,27 @@ pub async fn serve_web_inner(
         exec.clone(),
     ));
 
-    let mut srv = {
-        let builder = ViamServerBuilder::new(
-            mdns,
-            cloned_exec,
-            client_connector,
-            app_config,
-            max_webrtc_connection,
-            network,
-        )
-        .with_webrtc(webrtc)
-        .with_periodic_app_client_task(Box::new(RestartMonitor::new(|| unsafe {
-            crate::esp32::esp_idf_svc::sys::esp_restart()
-        })));
-        #[cfg(feature = "data")]
-        let builder = if let Some(task) = data_sync_task {
-            builder.with_periodic_app_client_task(Box::new(task))
-        } else {
-            builder
-        };
-        builder.build(&cfg_response).unwrap()
-    };
+    let mut srv = ViamServerBuilder::new(
+        mdns,
+        cloned_exec,
+        client_connector,
+        app_config,
+        max_webrtc_connection,
+        network,
+    )
+    .with_webrtc(webrtc)
+    .with_periodic_app_client_task(Box::new(RestartMonitor::new(|| unsafe {
+        crate::esp32::esp_idf_svc::sys::esp_restart()
+    })))
+    .build(&cfg_response)
+    .unwrap();
 
-    futures_lite::future::zip(Box::pin(srv.serve(robot)), data_future).await;
-}
+    // Attempt to cache the config for the machine we are about to `serve`.
+    if let Err(e) = storage.store_robot_configuration(cfg_response.config.unwrap()) {
```

Comment:
As above, would you be OK with having `store_robot_configuration` drop redundant writes by doing a read first?

Reply:
I think this is the way. In this particular case we should consider storing an hash (FNV32) of the config for faster comparison instead of doing a byte by byte.

Reply:
https://viam.atlassian.net/browse/RSDK-8120